### PR TITLE
feat: Implement Task::new_with_id method to create task with predefined Id

### PR DESCRIPTION
**Pull request by Artem468**
Hi, I think it would be useful if there was an option to also set your own task id. I've modified the code a bit as I see it for myself

### CODE REVIEW

**Comment by alex-karpenko** [comment]
Hello @Artem468 ,
Good idea, i think it will be useful.
But I would preffer to keep internals of `TaskId` and especially `JobId` hidden because Uuid is just an internal representation of it and potentially may be changed or extended in future.
There if From trait implemented  for Uuid->TaskId, so my suggestion is to use it.
Let me few minutes to prepare changes to the PR.

**Comment by Artem468** [comment]
I'm also thinking how I could get information about the task within the task itself. I have a project where I use "sacs" and there is a need to retrieve from Mutex<HashMap> data that was put by taskId or any other identifier that could be found in the task. Any ideas how to do this?

**Comment by alex-karpenko** [comment]
TaskId is constant value within a particular task, but JobId is internal id of particular run of the task, so it varies.
With yours proposed changes you can just move clone of the TaskId into job closure. Just rough example:
```rust
let task1_id = TaskId::default();
let task_id = task1_id.clone();

let task1 = Task::new_with_id(schedule.clone(), move |id| {
    let task_id = task_id.clone();
    Box::pin(async move {
        println!("task_id={task_id}"); // Here we have task_id which is constant within task
        // Actual async workload here
        tokio::time::sleep(Duration::from_secs(1)).await;
        // ...
        println!("Job {id} finished.");
        })
    }, task1_id);
```

File: src/task.rs [hunk 1]
```diff
@@ -58,6 +58,56 @@ impl Task {
         }
     }
 
+    /// Create a new Task with a specified schedule, job function TaskId.
+    ///
+    /// This method is useful if you need to know T`askId`` before task scheduled,
+    /// or need to use TaskId within job context: for particular task it's TaskId is constant value,
+    /// but JobId varies for each task run.
```
Comment (alex-karpenko) [inline src/task.rs:65]:
```suggestion
    /// Create a new `Task` with a specified schedule, job function and `TaskId`.
    ///
    /// This method is useful if you need to know `TaskId` before task scheduled,
    /// or need to use `TaskId` within job context: for particular task it's TaskId is constant value,
    /// but JobId varies for each task run.
```

**Review by alex-karpenko — APPROVED** [review]

**Pull request merged**